core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -360,6 +376,28 @@ private void maybeLogHeapDetails() {
     }
 
 
+    /**
+     * scans the node paths and loads existing metaData file. If not found a new meta data will be generated
+     * and persisted into the nodePaths
+     */
+    private static NodeMetaData loadOrCreateNodeMetaData(Settings settings, ESLogger logger,
+                                                         NodePath... nodePaths) throws IOException {
+        List<Path> pathList = Arrays.stream(nodePaths).map(np -> np.path).collect(Collectors.toList());
+        final Path[] paths = pathList.toArray(new Path[pathList.size()]);
```

Comment:
The above two lines can be simplified to
`final Path[] paths = Arrays.stream(nodePaths).map(np -> np.path).toArray(Path[]::new);`